model/task_lifecycle.go

```diff
@@ -1682,30 +1672,6 @@ func doRestartFailedTasks(tasks []string, user string, results RestartResults) R
 	return results
 }
 
-// CheckAndBlockSingleHostTaskGroup blocks a single-host task group if the given
-// task is part of one and has finished (or is currently finishing) with a
-// failed status. This is a best-effort attempt, so if it fails, it will just
-// log the error.
-func CheckAndBlockSingleHostTaskGroup(t *task.Task, status string) {
```

Comment:
We would have to keep this logic if we wanted backward compatibility with existing versions that don't have the dependencies set up. Is it worth keeping this for backward compatibility? I don't have strong opinions here.

Reply:
Hmm, yeah I think we should do that actually. I'll make a ticket to remove it later